skxray/speckle_analysis.py

```diff
@@ -0,0 +1,317 @@
+
+    Returns
+    -------
+    roi_pix : dict
+        intensities of the ROI's of the labeled array according
+        to the pixel list
+        {ROI 1 : intensities of the pixels of ROI 1, ROI 2 : intensities of
+         the pixels of ROI 2}
+    """
+
+    if labels.shape != image.shape:
+        raise ValueError("Shape of the image data should be equal to"
+                         " shape of the labeled array")
+
+    return {n: image[labels == n] for n in range(1, np.max(labels)+1)}
+
+
+def time_series(number=2, number_of_images=50):
+    """
+    This will provide the geometric series for the integration.
+    Last values of the series has to be less than or equal to number
+    of images
+    ex: number_of_images = 100
+    number = 2, time_series =  1, 2, 4, 8, 16, 32, 64
+    number = 3, time_series =  1, 3, 9, 27, 81
+
+    Parameters
+    ----------
+    number : int, optional
+        time steps for the integration
+
+    number_of_images : int, optional
+        number of images
+
+    Return
+    ------
+    time_series : list
+        time binning
+
+    Note
+    ----
+    :math ::
+     a + ar + ar^2 + ar^3 + ar^4 + ...
+
+     a - first term in the series
+     r - is the common ratio
+    """
+
+    time_series = [1]
+
+    while time_series[-1]*number < number_of_images:
+        time_series.append(time_series[-1]*number)
+    return time_series
+
+
+def mean_intensity_sets(images_set, labels):
+    """
+    Mean intensities for ROIS' of the labeled array for different image sets
+
+    Parameters
+    ----------
+    images : array
+        iterable of 4D arrays
+        shapes is: (len(images_sets), )
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_int_labels : dict
+        average intensity of each ROI as a dictionary
+        shape len(images_sets)
+        eg: 2 image sets,
+        {image set 1 : (len(images in image set 1), number of labels),
+        image set 2 : (len(images in image set 2), number of labels)}
+
+    """
+    return {n+1: mean_intensity(images_set[n],
+                                labels) for n in range(len(images_set))}
+
+
+def mean_intensity(images, labels):
+    """
+    Mean intensities for ROIS' of the labeled array for set of images
+
+    Parameters
+    ----------
+    images : array
+        iterable of 2D arrays
+        dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_intensity : array
+        mean intensity of each ROI for the set of images as an array
+        shape (len(images), number of labels)
+
+    """
+    if labels.shape != images[0].shape[0:]:
+        raise ValueError("Shape of the images should be equal to"
+                         " shape of the label array")
+
+    index = np.unique(labels)[1:]
```

Comment:
You do not want want to do this on the uniqueified labels, as this means you may not be able to lookup in the output the mean of a given ROI.